server/src/main/java/org/elasticsearch/index/IndexVersion.java

```diff
@@ -124,7 +124,7 @@ public static IndexVersion current() {
     }
 
     public boolean isLegacyIndexVersion() {
-        return before(IndexVersions.MINIMUM_COMPATIBLE);
+        return before(IndexVersions.MINIMUM_READONLY_COMPATIBLE);
```

Comment:
I wonder about this. Why do we need MINIMUM_READONLY_COMPATIBLE in 8.x if it's the same as MINIMUM_COMPATIBLE? It seems like the code in 8.x could be much simpler, although it would diverge from main? Anything I am missing?

Reply:
No you're not missing anything. Keeping the code similar on both branches helps for backporting changes and I suspect we will have more things to backport in a short future.

Reply:
Ok, up to you!